docs/07-lifecycle.md

```diff
@@ -2,8 +2,54 @@
 ## Development
 ## Submit to Marketplace
 ## Provision
+### App Bundle
+App bundle contains `manifest.json` file, `static/` folder and one or several lambda function bundles.
+- `static/` folder may contain static files app needs. They are automatically provisioned and stored in the AWS S3 bucket. 
+- `manifest.json` file contains details about the app such as appID, appVersion, appType(http or an AWS app), requested permissions, requested locations, and information about the functions.
+- Each of the lambda function bundles is a valid and runnable AWS lambda function, which are provisioned in the AWS by the Gitlab pipeline. 
+
+### Provisioning In The 3rd Party AWS
+Provisioning in the third party AWS cloud environment is done by the **appsctl** which reads appropriate AWS credentials from environment variables:
+
+`APPS_PROVISION_AWS_ACCESS_KEY`
+
+`APPS_PROVISION_AWS_SECRET_KEY`
+
+We need an app bundle to provision an app. The bundle might be provisioned from the local disk, from S3(not implemented yet) or from some URL(not implemented yet). Provisioning consists of 3 parts:
+
+1. Creating the lambda functions with appropriate policies
+2. Storing static assets in the dedicated S3 bucket.
+3. Storing app’s manifest file in the same dedicated S3 bucket.
+
+AWS Lambda functions have semantic names, which means that a function described in the manifest.json file translates to AWS as `$appID_$appVersion_$functionName` to avoid collisions with other apps' or other versions' functions. And **appsctl** provisions lambda functions using this name.
+Dedicated S3 bucket name is stored in the environment variable 
+
+`MM_APPS_S3_BUCKET`
+
+which stores all apps' static assets and manifest files. 
+
+
+All files in the static folder of the bundle are considered to be the app's static assets. And are stored in the above mentioned bucket. Stored assets also have semantic keys and are generated using the rule - `static/$appID_$appVersion/filename`. 
+
+
+The `manifest.json` file of an app is stored in the same S3 bucket as well with the key - `manifests/$appID_$appVersion.json`.
+
+### Provisioning In Mattermost AWS Cloud
+To be provisioned in AWS mattermost cloud an app bundle is uploaded to the specific S3 bucket(`mattermost-apps-bucket`). On a new app release bundle is created by the circleCI and uploaded to S3(the same way plugins do right now). After that apps provisioning pipeline, running on GitlabCI, automatically detects the upload, creates appropriate lambda functions and assets the same way the **appsclt** does for the third party accounts. To generate lambda function names and asset keys apps provisioning pipeline needs additional variables for the terraform. Those are generated by the provisioning pipeline using the
```

Comment:
"(the same way plugins do right now)" - propose we remove this - the audience will be the developer that might not have our context.
"On a new app release bundle" - propose to describe in more depth the process to get here.

Reply:
I think app release process could be determined elsewhere